dex-it/src/test/scala/com/wavesplatform/it/MatcherSuiteBase.scala

```diff
@@ -70,17 +71,18 @@ trait MatcherSuiteBase
     super.afterAll()
   }
 
-  def createAccountWithBalance(balances: (Long, Asset)*): KeyPair = {
+  def mkAccountWithBalance(balances: (Long, Asset)*): KeyPair = {
     val account = KeyPair(ByteStr(s"account-test-${ThreadLocalRandom.current().nextInt()}".getBytes(StandardCharsets.UTF_8)))
 
     balances.foreach {
-      case (balance, asset) => {
-        assert(
-          wavesNode1.api.balance(alice, asset) >= balance,
-          s"Bob doesn't have enough balance in ${asset.toString} to make a transfer"
-        )
-        broadcastAndAwait(mkTransfer(alice, account.toAddress, balance, asset))
-      }
+      case (balance, asset) =>
+        asset.fold { scalatest.Assertions.succeed } { issuedAsset =>
```

Comment:
1. Please use pattern matching instead of fold (there are reasons in the chat)
2. We are checking a pre-condition for issued assets, but not for Waves. This is not fair :)